tools/wptrunner/wptrunner/formatters/wptscreenshot.py

```diff
@@ -1,15 +1,33 @@
+import requests
 from mozlog.structured.formatters.base import BaseFormatter
 
 
 class WptscreenshotFormatter(BaseFormatter):
     """Formatter that outputs screenshots in the format expected by wpt.fyi."""
 
+    # TODO(Hexcles): Make this configurable.
+    API = "https://staging.wpt.fyi/api/screenshots/hashes"
+
     def __init__(self):
         self.cache = set()
 
     def suite_start(self, data):
-        # TODO: ask wpt.fyi for known hashes.
-        pass
+        params = {}
+        run_info = data.get("run_info", {})
+        if "product" in run_info:
+            params["browser"] = run_info["product"]
+        if "browser_version" in run_info:
+            params["browser_version"] = run_info["browser_version"]
+        if "os" in run_info:
+            params["os"] = run_info["os"]
+        if "os_version" in run_info:
+            params["os_version"] = run_info["os_version"]
+        try:
+            r = requests.get(self.API, params=params)
+            r.raise_for_status()
+            self.cache = set(r.json())
+        except (requests.exceptions.RequestException, ValueError):
+            pass
```

Comment:
Although the errors are non-fatal, I do want to surface them somehow. But how do we elegantly log something from within a log formatter?